examples/java/src/test/java/org/apache/beam/examples/cookbook/MaxPerKeyExamplesTest.java

```diff
@@ -66,22 +71,27 @@
   private static final TableRow resultRow1 =
       new TableRow().set("month", 6).set("max_mean_temp", 85.3);
   private static final TableRow resultRow2 =
+      new TableRow().set("month", 6).set("max_mean_temp", 45.3);
+  private static final TableRow resultRow3 =
       new TableRow().set("month", 7).set("max_mean_temp", 75.4);
 
+  @Rule public TestPipeline p = TestPipeline.create();
+
   @Test
-  public void testExtractTempFn() throws Exception {
-    DoFnTester<TableRow, KV<Integer, Double>> extractTempFn = DoFnTester.of(new ExtractTempFn());
-    List<KV<Integer, Double>> results = extractTempFn.processBundle(TEST_ROWS);
-    Assert.assertThat(results, CoreMatchers.hasItem(kv1));
-    Assert.assertThat(results, CoreMatchers.hasItem(kv2));
-    Assert.assertThat(results, CoreMatchers.hasItem(kv3));
+  @Category(ValidatesRunner.class)
```

Comment:
I don't think we need to annotate these tests with `ValidatesRunner.class` category. 
@kennknowles could you confirm this?

Reply:
Hello Alexey. As I understand this annotation is used to check that test's author write pipeline.run
(p.run().waitUntilFinish();) in the end of the test. It has > 300 usages in all project. I think it is necessary here.

What should I do with this tests?
I can rewrite it like this:

`  @Rule public TestPipeline p = TestPipeline.create()`
`          .enableAbandonedNodeEnforcement(false)`
`         .enableAutoRunIfMissing(true);`
` @Test` public void testExtractTempFn() {`
`    PCollection<...> results = ...;`
`    PAssert.that(results).containsInAnyOrder(ImmutableList.of(kv1, kv2, kv3));`
`  }`

This approach will start pipeline automatically in the end of test (as I think)

Reply:
I think it's ok for now